### ISSUE TITLE
Problems with queries

### ISSUE THREAD

**Issue by Hemualliug** [ld4mbse/oslc4tdb#5]
Hello Mr @isccarrasco 

I sent you a message a few months ago and your answers helped me a lot but my project with SodiusWillert is not finished and I still have hard times to make my queries work. 

So I'm still using the same SCHACL Schema for creating my DogGraph2,

```
@prefix rdf: <http://www.w3.org/1999/02/22-rdf-syntax-ns#> .
@prefix rdfs: <http://www.w3.org/2000/01/rdf-schema#> .
@prefix sh: <http://www.w3.org/ns/shacl#> .
@prefix xsd: <http://www.w3.org/2001/XMLSchema#> .
@prefix oslc: <http://open-services.net/ns/core#> .
@prefix foaf: <http://xmlns.com/foaf/0.1/> .
@prefix dcterms: <http://purl.org/dc/terms/> .
<urn:PersonShape> a sh:NodeShape ;
 sh:targetClass foaf:Person ;
 rdfs:label "Person" ;
 sh:property [
     sh:path foaf:name ;
     sh:nodeKind sh:Literal ;
     sh:datatype xsd:string ;
 ] ;
.
<urn:DogShape> a sh:NodeShape ;
 sh:targetClass foaf:Dog ;
 rdfs:label "Dog" ;
 sh:property [
     sh:path foaf:familyName ;
     sh:nodeKind sh:Literal ;
     sh:datatype xsd:string ;
 ] ;
.
```
Then I'm using the Creation Factory endpoint so as to insert 2 resources as follows:

```
POST http://localhost:8080/oslc4tdb/oslc/MyStore/DogGraph2/Dog
Body: 
<rdf:RDF xmlns:rdf="http://www.w3.org/1999/02/22-rdf-syntax-ns#"
         xmlns:foaf="http://xmlns.com/foaf/0.1/">
	<rdf:Description rdf:about="">
		<foaf:familyName>Fido</foaf:familyName>
		<rdf:type rdf:resource="http://xmlns.com/foaf/0.1/Dog" />
	</rdf:Description>
</rdf:RDF>

Content-Type: application/rdf+xml
Slug: 1
```
 and my second resource:
```
POST http://localhost:8080/oslc4tdb/oslc/MyStore/DogGraph2/Dog
<rdf:RDF xmlns:rdf="http://www.w3.org/1999/02/22-rdf-syntax-ns#"
         xmlns:foaf="http://xmlns.com/foaf/0.1/">
	<rdf:Description rdf:about="">
		<foaf:familyName>Ben</foaf:familyName>
		<rdf:type rdf:resource="http://xmlns.com/foaf/0.1/Dog" />
	</rdf:Description>
</rdf:RDF>

Content-Type: application/rdf+xml
Slug: 2
```
I'm then making a query in order to filter the Dog whose familyName is Ben for example with the following query:

```
GET http://localhost:8080/oslc4tdb/oslc/MyStore/DogGraph2/Dog?oslc.prefix=foaf=<http://xmlns.com/foaf/0.1/>&oslc.where=foaf:familyName="Ben"
```
So, I expect to have an answer in which I get the resource : http://localhost:8080/oslc4tdb/oslc/MyStore/DogGraph2/Dog/2
which is the resource in which the Dog's familyName is Ben. Instead I got the two "Dogs" as a response.

I don't understand because as think my query is well-formed but maybe I'm not using the oscl.where correctly but I followed and read the OSCL Query specification you suggested me in our former discussion.

Thank you,

Sincerely,

Guillaume H.

**Comment by axelreichwein** [comment]
Hello @Hemualliug does the last commit fix the issue?

**Comment by Hemualliug** [comment]
Hello @axelreichwein ,

I pulled your last commit but it still doesn't work, I still have the two dogs as a response for my query.
I then tried to re-download your project and re-build the war-archive, but now I can't build the war-archive with ``` mvn package``` .

The 12 tests failed. One of the message is:
``` 2021-05-05 10:35:24 [main] WARN  org.apache.jena.tdb.TDB - System file mode already determined - setting it has no effect``` 
and 
``` java.lang.NullPointerException``` 

Even if I tried to skip the tests with: ``` mvn package -DskipsTests ```, I can't get the war archive.

Regards,

GH
